### PR TITLE
Dbt debug changes

### PR DESCRIPTION
**Pull request by TapasSenapati**
Testing 

**incorrect username/password**
Connection:
  host: https://tx4lvf8t.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/dex/api/v1
  port: 443
  cluster: None
  endpoint: None
  schema: dbtbmstest
  organization: 0
Http Error:  401 Client Error: Unauthorized for url: https://service.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/gateway/authtkn/knoxtoken/api/v1/token
  Connection test: [ERROR]

==============================================
**incorrect authpoint**

Connection:
  host: https://tx4lvf8t.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/dex/api/v1/
  port: 443
  cluster: None
  endpoint: None
  schema: dbtbmstest
  organization: 0
Connection Error : HTTPSConnectionPool(host='service.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site1', port=443): Max retries exceeded with url: /gateway/authtkn/knoxtoken/api/v1/token (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x11085f2e0>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
  Connection test: [ERROR]

==============================================

**Incorrect hostname:**

Connection:
  host: https://tx4lvf8t.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/dex/api/v/
  port: 443
  cluster: None
  endpoint: None
  schema: dbtbmstest
  organization: 0
RequestException:  404 Client Error: Not Found for url: https://tx4lvf8t.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/dex/api/v/resources
  Connection test: [ERROR]

1 check failed:
dbt was unable to connect to the specified database.
The database returned the following error:

  >Runtime Error
  Runtime Error
    Please verify hostname in your dbt profiles file.

==================
**success** 

Configuration:
  profiles.yml file [OK found and valid]
  dbt_project.yml file [OK found and valid]

Required dependencies:
 - git [OK found]

Connection:
  host: https://tx4lvf8t.cde-bhs8thz9.ciadev.cna2-sx9y.cloudera.site/dex/api/v1/
  port: 443
  cluster: None
  endpoint: None
  schema: dbtbmstest
  organization: 0
  Connection test: [OK connection ok]

All checks passed!

### CODE REVIEW

**Review by tovganesh — DISMISSED** [review]
Looks neat

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 1]
```diff
@@ -325,6 +325,25 @@ def __init__(self, base_api_url, access_token, api_header) -> None:
...
+        def inner_function(*args, **kwargs):
+            res = None
+            try:
+                res = func(*args, **kwargs)
+                if res.status_code not in range(200, 300):
+                    raise dbt.exceptions.DbtProfileError(
+                        "Error communicating with cde spark host."
+                    )
+                res.raise_for_status()
+            except requests.exceptions.RequestException as err:
```
Comment (tovganesh) [inline dbt/adapters/spark_cde/cdeapisession.py:340]: do we re-raise the exception here ?

**Review by sujitkp-blr — APPROVED** [review]
looks good.

File: dbt/adapters/spark_cde/cdeapisession.py (hunk 1 above)
Comment (TapasSenapati) [reply dbt/adapters/spark_cde/cdeapisession.py]: cleaned it.

**Review by tovganesh — APPROVED** [review]

**Pull request merged**